### PR TITLE
kpt-ize the foundation package and add README

### PR DESCRIPTION
**Pull request by janetkuo**
@morgante @viglesiasce

### CODE REVIEW

File: krm/0_foundation/Kptfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: 0_foundation
```
Comment (morgante) [inline krm/0_foundation/Kptfile:4]:
```suggestion
  name: cicd_foundation
```

File: krm/0_foundation/Kptfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+packageMetadata:
+  shortDescription: Anthos CI/CD foundation layer
+openAPI:
+  definitions:
+    io.k8s.cli.setters.namespace:
+      x-k8s-cli:
+        setter:
+          name: namespace
+          value: YOUR_NAMESPACE
```
Comment (morgante) [inline krm/0_foundation/Kptfile:13]: I don't think we're meant to provide a value upfront? Also this should be marked as required.

Comment (janetkuo) [reply krm/0_foundation/Kptfile:13]: Is there a way to create a kpt setter without a value? Value is required in `kpt cfg create-setter`.

Comment (phanimarupaka) [reply krm/0_foundation/Kptfile:13]: Value is required to create a setter. what is the use-case for not creating setter with value ? In this scenario, you can have setter with value YOUR_NAMESPACE and mark it as required. 

https://googlecontainertools.github.io/kpt/guides/producer/setters/#marking-a-field-as-required

Comment (janetkuo) [reply krm/0_foundation/Kptfile]: Done - marking it as required

**Pull request merged**